tiledb/sm/group/group.h

```diff
@@ -101,16 +98,16 @@ class Group {
    *
    * @param uri The address of the group to be deleted.
    * @param recursive True if all data inside the group is to be deleted.
+   * @param close True if the group should also be closed after deletion.
    */
-  void delete_group(const URI& uri, bool recursive = false);
+  void delete_group(const URI& uri, bool recursive = false, bool close = true);
```

Comment:
This function signature will trigger a `clang-tidy` warning that the arguments can easily be swapped. `clang-tidy` is right. Random clusters of booleans at the end of function signatures make for hard-to-read code.

* Rename this function `delete_group_internal` and make it private.
  * Document is meticulous detail why we need two behaviors. Requiring both is a transitional measure, and the documentation should state that.
* Implement `delete_group` with the same semantics by forwarding to `delete_group_internal(..., true)`.
* Create a new function `delete_group_and_keep_open` by forwarding to `delete_group_internal(..., false)`. It's an intentionally ugly name to cope with some not-great code.